DHI inline patch (cassandra 4.1): bump ipaddress for CVE-2023-50570

Upstream fixed the invalid-radix infinite loop in ipaddress 5.4.1:
https://github.com/seancfoley/IPAddress/releases/tag/v5.4.1
https://github.com/seancfoley/IPAddress/issues/118

diff --git a/build.xml b/build.xml
index bb5b50e..8d90e2c 100644
--- a/build.xml
+++ b/build.xml
@@ -745,7 +745,7 @@
             <exclusion groupId="org.hamcrest" artifactId="hamcrest"/>
           </dependency>
           <dependency groupId="org.hamcrest" artifactId="hamcrest" version="2.2" scope="test"/>
-          <dependency groupId="com.github.seancfoley" artifactId="ipaddress" version="5.3.3" />
+          <dependency groupId="com.github.seancfoley" artifactId="ipaddress" version="5.4.1" />
         </dependencyManagement>
         <developer id="adelapena" name="Andres de la Peña"/>
         <developer id="alakshman" name="Avinash Lakshman"/>
